src/common/tse.c

```diff
@@ -461,12 +462,12 @@ tse_task_prep_callback(tse_task_t *task)
 
 		D_FREE(dtc);
 
-		/** Task was re-initialized; break */
+		/** Task was re-initialized; */
 		if (!dtp->dtp_running && !dtp->dtp_completing)
-			return false;
+			ret = false;
```

Comment:
just confirm should it break here? although I don't have use case that must break.
actually seems in daos code, there is no use case with multiple prep cbs? you change it for upper layer use case?

Reply:
well the idea was not to break, because if we do, we end up not freeing the dtc and cause a memory leak when the task is completed. this happens in the sched.c test when run with valgrind.
first prep cb, just completes the test, the second prep cb never gets freed.

but yes ttere are cases where we have more than 1 prep cb in other use cases.